File: test-suite-python/src/test/python/micronaut/docs/ioc/introspection/Person.py
# tag::class[]
from dataclasses import dataclass

from micronaut.core.annotation import AccessorsStyle, Introspected


@dataclass
@Introspected
@AccessorsStyle(readPrefixes=[""], writePrefixes=[""])  # <1>
class Person:
    name: str  # <2>
    age: int  # <2>
# end::class[]

File: test-suite-scala/src/test/scala/io/micronaut/docs/ioc/introspection/Person.scala
package io.micronaut.docs.ioc.introspection

// tag::class[]
import io.micronaut.core.annotation.AccessorsStyle
import io.micronaut.core.annotation.Introspected

@Introspected
@AccessorsStyle(readPrefixes = Array(""), writePrefixes = Array("")) // <1>
class Person(private var currentName: String, private var currentAge: Int):

  def name(): String = currentName // <2>

  def name(name: String): Unit = // <2>
    currentName = name

  def age(): Int = currentAge // <2>

  def age(age: Int): Unit = // <2>
    currentAge = age
// end::class[]
